src/components/button/theme.ts

```diff
@@ -77,6 +77,10 @@ export default {
       },
     },
   },
+  buttonIconResponsive: {
+    button: 'hidden md:flex',
```

Comment:
With this approach, you don't need to know if the button is `flex`, `block` or whatever:

```suggestion
    button: 'max-md:hidden',
```